haskell-src/Concordium/Types/Execution.hs

```diff
@@ -2088,7 +2088,9 @@ instance S.Serialize RejectReason where
       rejectReason <- S.getInt32be
       contractAddress <- S.get
       receiveName <- S.get
-      parameter <- S.get
+      -- Get the parameter without validating its length. This is OK here, because the length has been checked before reaching this point.
+      -- Checking requires the protocol version as input, which is why it is skipped here.
```

Comment:
Perhaps add that "before reaching this point" means that this is always an outcome of a valid transaction.

Which means that validation has already happened, so we don't need to do it again.

Reply:
I've rephrased it, let me know if it is ok.